packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -176,5 +195,36 @@ export const coinjoinMiddleware =
             api.dispatch(storageActions.saveCoinjoinDebugSettings());
         }
 
+        // automatically pause/restore coinjoin session depending on current route
+        // device may be locked by another connect call, so check on LOCK_DEVICE action as well
+        if (action.type === ROUTER.LOCATION_CHANGE || action.type === SUITE.LOCK_DEVICE) {
+            const state = api.getState();
+            const { locks } = state.suite;
+            if (!locks.includes(SUITE.LOCK_TYPE.DEVICE) && !locks.includes(SUITE.LOCK_TYPE.UI)) {
+                if (
+                    state.router.settingsBackRoute.name === 'wallet-send' &&
+                    !isCoinjoinSessionBlockedGlobally(state)
+                ) {
+                    // restore all interrupted sessions upon leaving send form
+                    restoreSessionOfEligibleAccounts(state);
```

Comment:
When leaving the send form we restore all interrupted sessions because at that point we do not know which one we have stopped.

When entering send form, we only pause the session for the current account because sending does not affect any other account.

Reply:
I see, yes, resuming all interrupted accounts makes sense.

I am not sure if `restoreSessionOfEligibleAccounts` is the best name since sounds a bit vague meaning. But that is maybe just my feeling.

Reply:
Renamed to `restoreInterruptedCoinjoinSessions`.